doc/project/code_flow.rst

```diff
@@ -6,38 +6,58 @@ Code Flow and Branches
 Introduction
 ************
 
-The zephyr Git repository has three types of branches:
+The Zephyr Git repository has three types of branches:
 
 main
   Which contains the latest state of development
 
-topic-\*
-  Topic branches that are used for shared development of a new feature
+collab-\*
+  Collaboration branches that are used for shared development
+  of new features to be introduced into main branch when ready. Creating a new
+  collaboration branch requires a justification and a TSC approval. Collaboration branches
+  shall be based off the main branch and any changes developed in the collab
+  branch shall target the main development branch. For released versions of
+  Zephyr, the introduction of fixes and new features, if approved by the TSC,
+  shall be done using backport pull requests.
 
 vx.y-branch
   Branches which track maintenance releases based on a major
   release
 
-Development in topic branches before features go to mainline allows teams to
+Development in collaboration branches before features go to mainline allows teams to
 work independently on a subsystem or a feature, improves efficiency and
 turnaround time, and encourages collaboration and streamlines communication
 between developers.
 
-Changes submitted to a development topic branch can evolve and improve
+Changes submitted to a collaboration branch can evolve and improve
 incrementally in a branch, before they are submitted to the mainline tree for
 final integration.
 
 By dedicating an isolated branch to complex features, it's
 possible to initiate in-depth discussions around new additions before
 integrating them into the official project.
 
+Collaboration branches are ephemeral and shall be removed once the collaboration work
+has been completed. When a branch is requested, the proposal should include the
+following:
+
+* Define exit criteria for merging the Collaboration branch changes back into main.
+* Define a timeline for the expected life cycle of the branch. It is
+  recommended to select a Zephyr release to set the timeline. Extensions to
+  this timeline require TSC approval,  i.e. by which release it should be
+  removed.
+
+Collaboration branches shall not be used as forks of the main development tree.
```

Comment:
I would drop this sentence because "fork" is an overloaded and confusing term. Here it is used to describe the intention of (not) merging back. But "fork" in the Github sense is a purely technical term, it's only about just branching in a different clone to avoid tedious and error-prone permissions and namespaces. A _Github_ "fork" is neutral and says nothing about lifecycles and future intentions, the owner of a Github fork is free to use it in any way they want. It's also the opposite of a collaboration branch, can't be both at the same time.

The previous text is very clear and enough, this last sentence just muddies the water for people not experienced with the subtle and contradictory nuances of git and Github jargons.